plugin/forward/connect.go

```diff
@@ -91,6 +91,15 @@ func (p *Proxy) Connect(ctx context.Context, state request.Request, opts options
 		proto = state.Proto()
 	}
 
+	var err error
+	defer func() {
```

Comment:
Because there are two places (WriteMsg, ReadMsg) where the function returns early in case of error. The `defer` makes it so that in both cases the error is evaluated and the failure counter increased. I just thought it would be nicer than adding the same if-then-counter-increase logic twice. @miekg

Reply:
defer's are still considered slow, so I rather not add more of them in the hot path.